src/Components/Forms/src/FieldIdentifier.cs

```diff
@@ -94,42 +93,61 @@ private static void ParseAccessor<T>(Expression<Func<T>> accessor, out object mo
             accessorBody = unaryExpression.Operand;
         }
 
-        if (!(accessorBody is MemberExpression memberExpression))
+        switch (accessorBody)
         {
-            throw new ArgumentException($"The provided expression contains a {accessorBody.GetType().Name} which is not supported. {nameof(FieldIdentifier)} only supports simple member accessors (fields, properties) of an object.");
+            case MemberExpression memberExpression:
+                // Identify the field name. We don't mind whether it's a property or field, or even something else.
+                fieldName = memberExpression.Member.Name;
+                // Get a reference to the model object
+                // i.e., given a value like "(something).MemberName", determine the runtime value of "(something)",
+                if (memberExpression.Expression is ConstantExpression constantExpression)
+                {
+                    if (constantExpression.Value is null)
+                    {
+                        throw new ArgumentException("The provided expression must evaluate to a non-null value.");
+                    }
+                    model = constantExpression.Value;
+                }
+                else if (memberExpression.Expression != null)
+                {
+                    // It would be great to cache this somehow, but it's unclear there's a reasonable way to do
+                    // so, given that it embeds captured values such as "this". We could consider special-casing
+                    // for "() => something.Member" and building a cache keyed by "something.GetType()" with values
+                    // of type Func<object, object> so we can cheaply map from "something" to "something.Member".
+                    var modelLambda = Expression.Lambda(memberExpression.Expression);
+                    var modelLambdaCompiled = (Func<object?>)modelLambda.Compile();
+                    var result = modelLambdaCompiled();
+                    if (result is null)
+                    {
+                        throw new ArgumentException("The provided expression must evaluate to a non-null value.");
+                    }
+                    model = result;
+                }
+                else
+                {
+                    throw new ArgumentException($"The provided expression contains a {accessorBody.GetType().Name} which is not supported. {nameof(FieldIdentifier)} only supports simple member accessors (fields, properties) of an object.");
+                }
+                break;
+            case MethodCallExpression methodCallExpression when ExpressionFormatter.IsSingleArgumentIndexer(accessorBody):
+                fieldName = ExpressionFormatter.FormatIndexArgument(methodCallExpression.Arguments[0]);
+                model = GetModelFromIndexer(methodCallExpression);
```

Comment:
I *think* this means you've also implemented https://github.com/dotnet/aspnetcore/issues/41618. Is that right? If so would be great to mark that as done in this preview too!

Reply:
Yes